### PR TITLE
Fix misbehaviour with 256+ hooks on a same function

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

Might also crash sometimes.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: tests/ext/sandbox-regression/handle_many_hooks.phpt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+--TEST--
+The tracer should not crash when many hooks are installed
```
Comment (pierotibou) [inline tests/ext/sandbox-regression/handle_many_hooks.phpt:2]: Out of curiosity, did this test failed consistently before fixing?

File: ext/span.c [hunk 2]
```diff
@@ -184,12 +184,13 @@ void ddtrace_open_span(ddtrace_span_data *span) {
     ddtrace_set_global_span_properties(span);
 }
 
+// += 2 increment to avoid zval type ever being 0
```
Comment (pierotibou) [inline ext/span.c:187]: That looks like magic :)

File: tests/ext/sandbox-regression/handle_many_hooks.phpt (hunk 1 above)
Comment (bwoebi) [reply tests/ext/sandbox-regression/handle_many_hooks.phpt:2]: Yes.

File: ext/span.c (hunk 2 above)
Comment (bwoebi) [reply ext/span.c:187]: Yes, I don't like it very much, but it works.

**Review by morrisonlevi — APPROVED** [review]

**Pull request merged**